tests/test_stub_utils.py

```diff
@@ -0,0 +1,329 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+
+from pyfar import Signal
+
+import stub_utils
+
+
+def test_signal_stub_properties():
+    """ Test comparing properties of Signal stub
+    with actual Signal implementation.
+    """
+    time = np.ones((1, 1024))
+    freq = np.ones((1, 1024))
+    sampling_rate = 44100
+    fft_norm = 'none'
+
+    signal_stub = stub_utils.signal_stub(time, freq, sampling_rate, fft_norm)
+    stub_dir = dir(signal_stub)
+    signal_dir = dir(Signal(time, sampling_rate))
+
+    assert stub_dir.sort() == signal_dir.sort()
+
+
+def test_signal_stub_times():
+    """ Test times vector of Signal stub.
+    """
+    time = np.ones((1, 4))
+    freq = np.ones((1, 3))
+    sampling_rate = 1
+    fft_norm = 'none'
+    times = np.array([0., 1., 2., 3.])
+
+    signal_stub = stub_utils.signal_stub(time, freq, sampling_rate, fft_norm)
+
+    npt.assert_allclose(signal_stub.times, times, rtol=1e-10)
+
+
+def test_signal_stub_frequencies_even():
+    """ Test frequencies vector of Signal stub,
+    even number of samples.
+    """
+    time = np.ones((1, 4))
+    freq = np.ones((1, 3))
+    sampling_rate = 1
+    fft_norm = 'none'
+    frequencies = np.array([0., 0.25, 0.5])
+
+    signal_stub = stub_utils.signal_stub(time, freq, sampling_rate, fft_norm)
+
+    npt.assert_allclose(signal_stub.frequencies, frequencies, rtol=1e-10)
+
+
+def test_signal_stub_frequencies_odd():
+    """ Test frequencies vector of Signal stub,
+    odd number of samples.
+    """
+    time = np.ones((1, 5))
+    freq = np.ones((1, 3))
+    sampling_rate = 1
+    fft_norm = 'none'
+    frequencies = np.array([0., 0.2, 0.4])
+
+    signal_stub = stub_utils.signal_stub(time, freq, sampling_rate, fft_norm)
+
+    npt.assert_allclose(signal_stub.frequencies, frequencies, rtol=1e-10)
+
+
+def test_impulse_func_single_channel():
+    """Test generation of delta impulse, single channel.
+    """
+    n_samples = 4
+    fft_norm = 'none'
+    cshape = (1,)
+    delay = 0
+    time_truth = np.array([[1, 0, 0, 0]])
+    freq_truth = np.array([[1, 1, 1]], dtype=complex)
+
+    time, freq = stub_utils.impulse_func(delay, n_samples, fft_norm, cshape)
+
+    npt.assert_allclose(time, time_truth, rtol=1e-10)
+    npt.assert_allclose(freq, freq_truth, rtol=1e-10)
+
+
+def test_impulse_func_rms():
+    """Test generation of delta impulse,
+    RMS FFT normalization.
+    """
+    n_samples = 6
+    cshape = (1,)
+    delay = 0
+    fft_norm = 'rms'
+    freq_truth = 1 / n_samples * np.array(
+                        [[1, np.sqrt(2), np.sqrt(2), 1]], dtype=complex)
```

Comment:
E126... also in a few more lines

Reply:
I think I missed something here, how would this be written correctly?

Reply:
This was over-indented. The correct indention would have been 4 spaces.